### DISCUSSION TITLE
Any tools to visualize the correspondences in Super4PCS?

### DISCUSSION THREAD

**Discussion by Shubodh** [STORM-IRIT/OpenGR#96] [Ideas]
Are there any existing tools with which Super4PCS can be recompiled to visualize the bases/correspondences that the algorithm is finding? If not, could you give advice as to how to go about it, perhaps by using PCL wrapper?

Also, Super4PCS is doing a great job of giving good rotation values even in situations with very less overlap, but translation values are way off. Could you suggest how this problem can be solved? I will post the results on the point clouds I used if you need me to.

Thank you.

**Comment by nmellado** [comment]
Hi @Shubodh ,

> Are there any existing tools with which Super4PCS can be recompiled to visualize the bases/correspondences that the algorithm is finding? If not, could you give advice as to how to go about it, perhaps by using PCL wrapper?

There is currently no such things directly in the library, but we are currently working on a standalone software providing a GUI to the library. This should simplify the parameter setting, and add the kind of visualization tools you mentioned. Early releases are planned for this summer. 
This will be announced here: https://www.irit.fr/STORM/site/3d-point-cloud-registration/

> Also, Super4PCS is doing a great job of giving good rotation values even in situations with very less overlap, but translation values are way off. Could you suggest how this problem can be solved? I will post the results on the point clouds I used if you need me to.

That's the first time I get this time of feedback, could you please share data leading to this behavior (input+parameters) ?
Also, could you please create a dedicated issue to this second point ? So we can keep this thread alive for the visualization tool,

Cheers,

**Comment by Shubodh** [comment]
Thanks a lot for the information @nmellado. 

I have emailed you the data.
Sure, I will create another issue.
